Assets/Scripts/Interactable/InteractableObject.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class InteractableObject : MonoBehaviour, IInteractableObject
+{
+    public GameObject GamePrefab;
+    private string hoverName;
+
+    public int AmountTries = 0;
+
+    private void Start()
+    {
+        DetermineObject();
+    }
+
+    public void DoAction(GameObject player)
+    {
+        MiniGameManager.Instance.StartGame(GamePrefab);
+        MiniGameManager.Instance.InteractableObject = this;
+    }
+
+    public string GetHoverName() => hoverName;
+
+    private void DetermineObject()
+    {
+        string tag = gameObject.tag;
+
+        switch(tag)
+        {
+            case "StreetLamp":
+                hoverName = "Upgrade";
+                break;
+            case "Tree":
+                hoverName = "Plant";
+                break;
+            case "ManHole":
+                hoverName = "Clean";
+                break;
+            case "SolarPanel":
+                hoverName = "Fix";
+                break;
+            default:
+                hoverName = "Interact";
+                break;
+        }
+    }
+
+}
```

Comment:
We can put the type of the object in an enum like that: 
![image](https://user-images.githubusercontent.com/54850409/159562215-19f351ca-c2aa-449b-94d2-7323f1914816.png)

- That will make it more scalable, not having to look for strings to change all the time, and we can also avoid using tags (by adding a get;private set property) (The enum should be in its own file)

Reply:
@logoffro1  Thank you!! How would I use tags with enum?

Reply:
> @logoffro1 Thank you!! How would I use tags with enum?

You don't, half the idea of enums is to hopefully get rid of tags